src/bootstrap/bin/main.rs

```diff
@@ -16,17 +20,30 @@ fn main() {
     let mut build_lock;
     let _build_lock_guard;
     if cfg!(any(unix, windows)) {
+        // Get existing contents of lock file
         let path = config.out.join("lock");
+        let mut pid = Vec::new();
+        match std::fs::File::open(&path) {
+            Ok(mut lock_file) => {
+                t!(lock_file.read_to_end(&mut pid));
+            }
+            Err(_) => {} // Error is ignored, file is created later
+        };
+
         build_lock = fd_lock::RwLock::new(t!(std::fs::File::create(&path)));
+
         _build_lock_guard = match build_lock.try_write() {
-            Ok(lock) => lock,
+            Ok(mut lock) => {
+                t!(lock.write(&process::id().to_le_bytes()));
```

Comment:
I would probably stringify the pid, it'll make out of band manual inspection much easier.